src/pybind/cephfs/cephfs.pyx

```diff
@@ -1169,6 +1182,59 @@ cdef class LibCephFS(object):
                           st_atime=datetime.fromtimestamp(stx.stx_atime.tv_sec),
                           st_mtime=datetime.fromtimestamp(stx.stx_mtime.tv_sec),
                           st_ctime=datetime.fromtimestamp(stx.stx_ctime.tv_sec))
+    
+    def statx(self, path, flag):
+        """
+        Get a file's extended statistics and attributes.
+
+        :param path: the file or directory to get the statistics of.
+        :param flag: want bitfield of CEPH_STATX_* flags showing designed attributes.
+        """
+
+        self.require_state("mounted")
+        path = cstr(path, 'path')
+        if not isinstance(flag, int):
+            raise TypeError('flag must be a int')
+
+        cdef:
+            char* _path = path
+            statx stx
+            int _flag = flag
+            dict_result = dict()
+
+        with nogil:
+            ret = ceph_statx(self.cluster, _path, &stx, _flag, 0)
```

Comment:
The ceph_statx() flags parameter is always set to 0 here. I think it would be good to add a separate "flags" parameter to the statx python function so that you can also use this to stat a symlink. Eventually we may add support for AT_STATX_DONT_SYNC as well, and it would be good to be able to pass that in here in some fashion.

Reply:
updated.
Considering that there may be more than 2 flag flags, I added the flag parameter.